packages/esbuild/test/bundle/bundle.golden.txt

```diff
@@ -1,8 +1,53 @@
 (() => {
+  var __create = Object.create;
```

Comment:
whoa what happened here?

Reply:
I don't know. @mattem can you take a look at this golden bundle change?

Reply:
The import now comes via the linker I'm guessing.
Previously esbuild was picking up the `mjs` file from `bazel-out` where it's now finding the `js` in `node_modules`.
Ideally you'd set `resolveExtensions` to prefer the `mjs` over the `js`:

```
    args = {
        "resolveExtensions": [
            ".mjs",
            ".js",
        ],
    },
```
Perhaps we could make that the default again (it was previously, but we removed it to remove as much customization and use vanilla esbuild as much as possible).

This would then result in:
```
(() => {
  // node_modules/lib/index.mjs
  var NAME = "bazel";
  // bazel-out/darwin-fastbuild/bin/packages/esbuild/test/bundle/a.mjs
  console.log(`Hello ${NAME}`);
})();
//# sourceMappingURL=bundle.js.map
```

From the bundle metadata from your diff:
```
"inputs": {
    "node_modules/lib/index.js": {
      "bytes": 864,
      "imports": []
    },
    "bazel-out/darwin-fastbuild/bin/packages/esbuild/test/bundle/a.mjs": {
      "bytes": 520,
      "imports": [
        {
          "path": "node_modules/lib/index.js",
          "kind": "import-statement"
        }
      ]
    }
```

From HEAD on 4.x:
```
"inputs": {
    "bazel-out/darwin-fastbuild/bin/packages/esbuild/test/bundle/index.mjs": {
      "bytes": 395,
      "imports": []
    },
    "bazel-out/darwin-fastbuild/bin/packages/esbuild/test/bundle/a.mjs": {
      "bytes": 520,
      "imports": [
        {
          "path": "bazel-out/darwin-fastbuild/bin/packages/esbuild/test/bundle/index.mjs",
          "kind": "import-statement"
        }
      ]
    }
```

In esbuild, we generate the package mappings into a tsconfig file to ensure we pick them up from bazel-out, not node_modules. We could change this though.
https://github.com/bazelbuild/rules_nodejs/blob/4.x/packages/esbuild/esbuild.bzl#L36

Reply:
brilliant. worked as predicted. thanks!

not sure about making it default as this would only be an issue with ts_library which produces both right?